src/BlazorWebFormsComponents/BaseWebFormsComponent.cs

```diff
@@ -223,24 +212,29 @@ public ValueTask DisposeAsync()
 		#endregion
 
 		public bool LayoutTemplateRendered { get; set; } = false;
+		private static FieldInfo _renderFragmentField;
+		private const string BASEFRAGMENTFIELDNAME = "_renderFragment";
+		private const string PARENTCOMPONENTNAME = "ParentComponent";
+		//if you added ChildContent to this class, we could just nameof(ChildContent)
+		private const string CHILDCONTENTNAME = "ChildContent";
 
 		public BaseWebFormsComponent ()
 		{
-			RenderFragment _standardRender = builder2 => {
-				_hasPendingQueuedRender = false;
-				_hasNeverRendered = false;
-				BuildRenderTree(builder2);
-				// Include this to see it working
-				//builder2.AddContent(9999,"All your bases are loaded");
-			};
-			_renderFragment = builder =>
+			// Get Access to the ComponentBase field we need to wrap every component in a CascadingValue
+			_renderFragmentField ??= typeof(ComponentBase).GetField(BASEFRAGMENTFIELDNAME, BindingFlags.NonPublic | BindingFlags.Instance);
+
+			// Grab a copy of the default RenderFragment to go into the CascadingValue
+			var _standardRender = (RenderFragment)_renderFragmentField.GetValue(this);
+
+			// Override the default RenderFragment with our Special Sauce version
+			_renderFragmentField.SetValue(this, (RenderFragment)(builder =>
 				{
 					builder.OpenComponent(1, typeof(CascadingValue<BaseWebFormsComponent>));
-					builder.AddAttribute(2, "Name", "ParentComponent");
-					builder.AddAttribute(3, "Value", this);
-					builder.AddAttribute(4, "ChildContent", _standardRender);
+					builder.AddAttribute(2, nameof(CascadingValue<object>.Name), PARENTCOMPONENTNAME);
+					builder.AddAttribute(3, nameof(CascadingValue<object>.Value), this);
+					builder.AddAttribute(4, CHILDCONTENTNAME, _standardRender);
 					builder.CloseComponent();
-				};
+				}));
 		}
 

```

Comment:
This makes it more obvious what is going on I think: 
```suggestion
private void ParentWrappingBuildRenderTree(RenderTreeBuilder builder)
{
    builder.OpenComponent(1, typeof(CascadingValue<BaseWebFormsComponent>));
    builder.AddAttribute(2, nameof(CascadingValue<object>.Name), PARENTCOMPONENTNAME);
    builder.AddAttribute(3, nameof(CascadingValue<object>.Value), this);
    builder.AddAttribute(4, CHILDCONTENTNAME, _baseRenderFragment);
}
```